src/Http/Http/src/HttpBufferingOptions.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+
+namespace Microsoft.AspNetCore.Http
+{
+    public class HttpBufferingOptions
+    {
+        public string TempFileDirectory { get; set; } = Path.GetTempPath();
```

Comment:
How do you intend to account for `ASPNETCORE_TEMP`?

One pattern you could use is to leave TempFileDirectory null here and use PostConfigure to set a default if nobody else does. PostConfigure could also read `ASPNETCORE_TEMP` from config. See https://github.com/aspnet/AspNetCore/blob/master/src/Security/Authentication/Cookies/src/PostConfigureCookieAuthenticationOptions.cs

Reply:
PostConfigure checking for ` Environment.GetEnvironmentVariable("ASPNETCORE_TEMP") ??                              Path.GetTempPath()` when TempFileDirectory is null looks very clean. Will go this way 👍